nexus/server.go

```diff
@@ -13,9 +13,30 @@ import (
 	"net/url"
 	"strconv"
 	"strings"
+	"sync"
 	"time"
 )
 
+type handlerCtxKeyType struct{}
+
+var handlerCtxKey = handlerCtxKeyType{}
+
+type handlerCtx struct {
+	mu    sync.Mutex
+	links []Link
+}
+
+// AddHandlerLinks associates links with the current operation to be propagated back to the caller. This method
+// Can be called from an [Operation] handler Start method or from a [Handler] StartOperation method. The context
+// provided must be the context passed to the handler. This method may be called multiple times for a given handler,
+// each call appending additional links. Links will only be attached on successful responses.
+func AddHandlerLinks(ctx context.Context, links ...Link) {
```

Comment:
I should be able to see links that exist, remove links, etc. I would recommend at least a getter and setter if not access to a structure that contains links as a field.

Reply:
Sorry, I already merged, I can address in a followup PR if needed.

I think it's okay to expose a simple API for now, following https://pkg.go.dev/google.golang.org/grpc#SetHeader, and add more functionality later.